src/main/resources/archetype-resources/__rootArtifactId__/src/main/java/__pluginFormat__/internal/proxy/__pluginClass__ProxyFacetImpl.java

```diff
@@ -0,0 +1,223 @@
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.Context;
+import org.sonatype.nexus.repository.view.matchers.token.TokenMatcher;
+import org.sonatype.nexus.transaction.UnitOfWork;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.plugins.${pluginFormat}.internal.util.${pluginClass}PathUtils.*;
+import static org.sonatype.nexus.repository.storage.AssetEntityAdapter.P_ASSET_KIND;
+
+/**
+ * ${pluginClass} {@link ProxyFacet} implementation.
+ *
+ * @since 0.0.1
+ */
+@Named
+public class ${pluginClass}ProxyFacetImpl
+    extends ProxyFacetSupport
+    implements ${pluginClass}ProxyFacet
+{
+  private ${pluginClass}PathUtils ${pluginFormat}PathUtils;
+
+  private ${pluginClass}DataAccess ${pluginFormat}DataAccess;
+
+  @Inject
+  public ${pluginClass}ProxyFacetImpl(final ${pluginClass}PathUtils ${pluginFormat}PathUtils,
+                             final ${pluginClass}DataAccess ${pluginFormat}DataAccess)
+  {
+    this.${pluginFormat}PathUtils = checkNotNull(${pluginFormat}PathUtils);
+    this.${pluginFormat}DataAccess = checkNotNull(${pluginFormat}DataAccess);
+  }
+
+  // HACK: Workaround for known CGLIB issue, forces an Import-Package for org.sonatype.nexus.repository.config
+  @Override
+  protected void doValidate(final Configuration configuration) throws Exception {
+    super.doValidate(configuration);
+  }
+
+  @Nullable
+  @Override
+  protected Content getCachedContent(final Context context) {
+    AssetKind assetKind = context.getAttributes().require(AssetKind.class);
+    TokenMatcher.State matcherState = ${pluginFormat}PathUtils.matcherState(context);
+    switch (assetKind) {
+      case PACKAGES:
+        return getAsset(${pluginFormat}PathUtils.buildAssetPath(matcherState, INDEX_HTML));
+      case ARCHIVE:
+        return getAsset(${pluginFormat}PathUtils.buildAssetPath(matcherState, CHANNELDATA_JSON));
+      default:
+        throw new IllegalStateException("Received an invalid AssetKind of type: " + assetKind.name());
+    }
+  }
+
+  @TransactionalTouchBlob
+  //protected Content getAsset(final String assetPath) {
+  public Content getAsset(final String assetPath) {
+    StorageTx tx = UnitOfWork.currentTx();
+
+    Asset asset = ${pluginFormat}DataAccess.findAsset(tx, tx.findBucket(getRepository()), assetPath);
+    if (asset == null) {
+      return null;
+    }
+    return ${pluginFormat}DataAccess.toContent(asset, tx.requireBlob(asset.requireBlobRef()));
+  }
+
+  @Override
+  protected Content store(final Context context, final Content content) throws IOException {
+    AssetKind assetKind = context.getAttributes().require(AssetKind.class);
+    TokenMatcher.State matcherState = ${pluginFormat}PathUtils.matcherState(context);
+    switch (assetKind) {
+      case PACKAGES:
+        return putMetadata(content,
+            assetKind,
+            ${pluginFormat}PathUtils.buildAssetPath(matcherState, INDEX_HTML));
```

Comment:
Suggestion: Maybe make `INDEX_HTML` and `CHANNELDATA_JSON` more generic

Reply:
Yes it should! Your comment gets me going again in the right direction. I'm working on more generic examples in the template now. Thanks! (Stay tuned.)

Reply:
Start of more generic examples: 9c54d0a